### PR TITLE
HWP-365 Added post tag tests. Cleaned up functions.

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Wrote preliminary tests for Post Tags. 
- Cleaned up function files a bit. (spacing & .query use)

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/tests/api-deletePostTags.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        // Create some communities
+        await community.createCommunity(communityName, randomText, randomText, [{"tag": tag1, "count": 1}], loginResponse.body.token);
+
+        // Join communities
+        await community.joinCommunity(communityName, loginResponse.body.token);
+
+        // Create some posts
+        postResponse = await post.createPost(randomText, randomText, communityName, loginResponse.body.token);
+
+        // Tag that post
```
Comment (BradyMitch) [inline app/server/tests/api-deletePostTags.test.js:34]: You can only tag posts with the tags that have been set by the community creator.

File: app/server/tests/api-deletePostTags.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    afterAll(async ()=> {
+        await post.deleteAllPosts();
+        await community.deleteCommunity(communityName, loginResponse.body.token);
+        await auth.deleteUsers();
+    });
+
+    test('User can delete existing post tag.', async () => {
+        response = await post.deletePostTags(postResponse.body._id, loginResponse.body.token);
+        expect(response.status).toBe(204);
+        expect(response.body).not.toEqual(
```
Comment (BradyMitch) [inline app/server/tests/api-deletePostTags.test.js:47]: An unsuccessful tag would result in a 404, 403, or 400 status and does not include any json content.

File: app/server/tests/api-deletePostTags.test.js (hunk 1 above)
Comment (BradyMitch) [reply app/server/tests/api-deletePostTags.test.js:34]: Looks like you added tags after community creation in another test, but not this one.

Comment (dbarkowsky) [reply app/server/tests/api-deletePostTags.test.js:34]: The community is created in the beforeAll block, so it is all one user here who is registered, logged in, creates a community, joins it, posts, then tags that post before the first test case is even reached. A lot of set up just to test deletion. Tags were added to community during creation here. 

There was a test where I added a second tag to community, but that was case-specific to that test.

File: app/server/tests/api-deletePostTags.test.js (hunk 2 above)
Comment (dbarkowsky) [reply app/server/tests/api-deletePostTags.test.js:47]: Good point. I'll add a GET in there to return the tag array and check the not-expected tag against that.

**Pull request merged**